Chore: Intro to Storybook updates for 6.0 (react version) (#363)
chromaui/learnstorybook.com: content/intro-to-storybook/react/en/simple-component.md

// src/components/Task.stories.js

import React from 'react';

import Task from './Task';

export const actionsData = {
  onPinTask: { action: 'onPinTask' },
  onArchiveTask: { action: 'onArchiveTask' },

> };

Reviewer: Can we use argtypes for the actions?

I'm not sure what's better, either we could:
 - Define a global `argTypeRegex` in `preview.js` or
 - Add action annotations per argTypes in each component.

WDYT @jonniebigodes ?

Reviewer: Awesome!

- `component` -- the component itself,
- `title` -- how to refer to the component in the sidebar of the Storybook app,
- `excludeStories` -- exports in the story file that should not be rendered as stories by Storybook.
- `argTypes` -- specify the [args](https://storybook.js.org/docs/react/api/argtypes) behavior in each story.

To define our stories, we export a function for each of our test states to generate a story. The story is a function that returns a rendered element (i.e. a component with a set of props) in a given state---exactly like a [Stateless Functional Component](https://reactjs.org/docs/components-and-props.html).

As we have multiple permutations of our component, it's convenient to assign it to a `Template` variable. Introducing this pattern in your stories will reduce the amount of code you need to write and maintain.

<div class="aside">

> Take in consideration that `Template.bind({})` is a [standard JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind) technique for making a copy of a function. We recommend this technique, so each exported story can set its own properties on it.

Reviewer: ```suggestion
`Template.bind({})` is a [standard JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind) technique for making a copy of a function. We use this technique to allow each exported story to set its own properties, but use the same implementation.
```

</div>

> Arguments or [`args`](https://storybook.js.org/docs/react/writing-stories/args) for short, are extremely helpful for editing our components. They allow us to edit our components without restarting Storybook. Once the [`args`](https://storybook.js.org/docs/react/writing-stories/args) value changes so does the component.

Reviewer: ```suggestion
Arguments or [`args`](https://storybook.js.org/docs/react/writing-stories/args) for short, are extremely helpful for editing our components. They allow us to edit our components with the controls addon without restarting Storybook. Once the [`args`](https://storybook.js.org/docs/react/writing-stories/args) value changes so does the component.
```

module.exports = {
  stories: ['../src/components/**/*.stories.js'],
  addons: [
    '@storybook/addon-links',
    '@storybook/addon-essentials',
    '@storybook/preset-create-react-app',
  ],
};
```

After completing the change above, inside the `.storybook` folder, change the `preview.js` to the following:

```javascript
// .storybook/preview.js

import '../src/index.css';

export const parameters = {

>   actions: { argTypesRegex: '^on[A-Z].*' },

Reviewer: Oh! If we did this we don't need to add the actions args.

Reviewer: Yeah, perhaps we need some text in that spot to explain what's happening, WDYT?

Author: Added with the latest commit. Didn't want to go into it too much, gave a description of what are they and how they will be used. Also left a link pointing to the official documentation, should the reader wants to learn more about it.

Let me know if you're ok with it.
